### PR TITLE
add filter to backend and add sliding filter to frontend

### PR DESCRIPTION
**Pull request by mischaboldy**

### CODE REVIEW

File: client/src/app/main/machine/machine.component.ts [hunk 1]
```diff
@@ -41,9 +44,13 @@ export class MachineComponent implements OnInit {
         });
   }
 
-  private retrieveMachine(): void {
-    this._machineService.find().subscribe(
-      (machines: Machine[]) => this.machines = machines,
+  private retrieveMachines(filterInfo: any): void {
```
Comment (jpastena) [inline client/src/app/main/machine/machine.component.ts]: give a type

File: server/src/main/java/com/example/controller/MachineController.java [hunk 2]
```diff
@@ -24,6 +25,11 @@ public Machine getMachineById(@PathVariable String id) {
 		return this.machineService.getOneById(id);
 	}
 
+	@PostMapping("WithFilter")
+	public List<Machine> getAllWithFilter(@RequestBody MachineFilter filter) {
```
Comment (jpastena) [inline server/src/main/java/com/example/controller/MachineController.java]: @Valid

File: client/src/app/shared/forms/filter-machine-form/filter-machine-form.constant.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  applicationEnvironment: new FormControl(),
+  hostingEnvironment: new FormControl(),
+  functionality: new FormControl(),
+  operatingSystem: new FormControl(),
+  assetName: new FormControl(),
+  hostName: new FormControl(),
+  status: new FormControl()
+};
+
+
```
Comment (jpastena) [inline client/src/app/shared/forms/filter-machine-form/filter-machine-form.constant.ts]: double new line <.<

File: client/src/app/main/machine/filter/machine-filter.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  public sendFilter(): void {
+    this.onFilter.emit(this.machineFilterFormGroup.value);
+  }
+
+  public toggleMenuState() {
+    this.menuState = this.menuState === SlideStates.In ? SlideStates.Out : SlideStates.In;
+  }
+
+  public getScreenheight() {
```
Comment (maxiemwagenaar) [inline client/src/app/main/machine/filter/machine-filter.component.ts]: getScreenHeight()

**Pull request merged**